diff --git a/torch/headeronly/macros/Export.h b/torch/headeronly/macros/Export.h
index 282806f386f..7dba7ffe3bb 100644
--- a/torch/headeronly/macros/Export.h
+++ b/torch/headeronly/macros/Export.h
@@ -123,13 +123,13 @@
 // libtorch_cuda.so (where torch_cuda_cu and torch_cuda_cpp are a part of the
 // same api)

-#ifdef TORCH_CUDA_BUILD_MAIN_LIB
+/*#ifdef TORCH_CUDA_BUILD_MAIN_LIB
 #define TORCH_HIP_CPP_API C10_EXPORT
 #define TORCH_HIP_API C10_EXPORT
 #else
 #define TORCH_HIP_CPP_API C10_IMPORT
 #define TORCH_HIP_API C10_IMPORT
-#endif
+#endif*/

 #if defined(TORCH_HIP_BUILD_MAIN_LIB)
 #define TORCH_HIP_CPP_API C10_EXPORT
